### PR TITLE
Blog post on method interception

### PR DESCRIPTION
**Pull request by Soothsilver**

### CODE REVIEW

File: _posts/2020-08-21-interception.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+   }
+}
+```
+If you then annotate a method with `[Cache]`, the method body will only run once. Subsequent calls will only return the cached `knownResult` value.  
+
+This example doesn't take into account that you might want to call the method body again if the method was called on a different object, with different parameters or if enough time already elapsed.
+
+We have [a more advanced caching sample](https://samples.postsharp.net/f/PostSharp.Samples.CustomCaching/) as well.
+## [RunsOnCustomThread]
+Xunit normally runs tests on thread pool threads. But, in some unit tests for the PostSharp Threading library, I need to make sure that the unit test runs on a non-thread-pool thread.
```
Comment (gfraiteur) [inline _posts/2020-08-21-interception.md:102]: Rename to `RunOnOwnThread`? _Custom_ does not mean much here and other aspects don't have an `s`.

Comment (gfraiteur) [reply _posts/2020-08-21-interception.md:102]: i.e. there's a typo in the title

File: _posts/2020-08-21-interception.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+---
+layout: post 
+comments: true
+title: "Intercepting methods"
+permalink: /post/intercepting-methods.html
+author: "Petr H."
+published: false
+excerpt_separator: <!--more-->
```
Comment (gfraiteur) [inline _posts/2020-08-21-interception.md:8]: Not sure `excerpt_separator` works with our template. The developer said it does not.

File: _posts/2020-08-21-interception.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+```c#
+[Fact, RunOnCustomThread]
+public void TestThreadPoolOperation()
+{
+    // unit test here
+}
+```
+I had a guarantee that the unit test will run on its own thread.
+## Works on all methods
+PostSharp interception aspects work on all kinds of methods: you can intercept both instance and static methods, and both public and private methods. You don't need interfaces and the methods don't need to be virtual.
```
Comment (gfraiteur) [inline _posts/2020-08-21-interception.md:142]: You can add _unlike interceptors of dependency injection frameworks_

File: _posts/2020-08-21-interception.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
+---
+layout: post 
+comments: true
+title: "Intercepting methods"
```
Comment (gfraiteur) [inline _posts/2020-08-21-interception.md:4]: Discuss the title with Lejla - the series could be named 'Focus on PostSharp Community'. You could rename the method boundary article and include cross links to form a series.

File: _posts/2020-08-21-interception.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+permalink: /post/intercepting-methods.html
+author: "Petr H."
+published: false
+excerpt_separator: <!--more-->
+---
+**Method interception** is a technique where you annotate a method and then when it's called, an interceptor is executed _instead of_ the method body. PostSharp Community allows you to add such interceptors to your code.
+
```
Comment (gfraiteur) [inline _posts/2020-08-21-interception.md:11]: Say that PostSharp Community is free

**Review by gfraiteur — COMMENTED** [review]
Good in general. Just a few comments.

File: _posts/2020-08-21-interception.md (hunk 1 above)
Comment (Soothsilver) [reply _posts/2020-08-21-interception.md:102]: Fixed

File: _posts/2020-08-21-interception.md (hunk 2 above)
Comment (Soothsilver) [reply _posts/2020-08-21-interception.md:8]: It worked for me when I built it locally.

File: _posts/2020-08-21-interception.md (hunk 3 above)
Comment (Soothsilver) [reply _posts/2020-08-21-interception.md:142]: Added.

File: _posts/2020-08-21-interception.md (hunk 5 above)
Comment (Soothsilver) [reply _posts/2020-08-21-interception.md:11]: Added.

File: _posts/2020-08-21-interception.md (hunk 4 above)
Comment (Soothsilver) [reply _posts/2020-08-21-interception.md]: Discussed, updated article, updated title.

**Pull request merged**